clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -22,11 +22,11 @@
     private final String topic;
     private final int partition;
     private final long offset;
-    private final long timestamp;
+    private final Long timestamp;
     private final TimestampType timestampType;
-    private final long checksum;
-    private final int serializedKeySize;
-    private final int serializedValueSize;
+    private final Long checksum;
+    private final Integer serializedKeySize;
+    private final Integer serializedValueSize;
```

Comment:
Hmm... on second thought, I'm wondering if that is better. The null protects users from misusage of the field. We could also wrap these fields into an object (e.g. a consumer version of RecordMetadata).